### ISSUE TITLE
OpenVPN: client specific override to push a specified IP address

### ISSUE THREAD

**Issue by muchachagrande** [opnsense/core#7287]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [ x ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ x ] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Is your feature request related to a problem? Please describe.**

A clear and concise description of what the problem is including your motivation for the request,

Using OpenVPN, to be able to push a specified IP address to a client, the command ifconfig-push is used in the client specific overrides configuration with the IP address and the subnet mask.
Before, I was able to use this command in the advanced section, but this free text box has gone for security reasons.

**Describe the solution you like**

I wonder if this option can be added as a specific input box, only for this purpose, at the bottom of the client settings section of the client specific overrides menu.
The configuration command is, for example, "ifconfig-push 172.18.0.1 255.255.255.0", so the user could enter an IP address and a subnet mask in the input box.

**Describe alternatives you considered**

I don't know an alternative way of pushing a fixed IP address to a client with OpenVPN.

**Additional context**
An example of the "Client specific overrides" menu with the new option at the bottom:

![ClientOverrideFixedIP](https://github.com/opnsense/core/assets/20299175/27177a01-267c-4202-b981-1c711a9566c5)

Of course, thank you very much and cheers

**Comment by AdSchellevis** [comment]
This is not how it's supposed to work, see https://docs.opnsense.org/manual/vpnet.html#client-specific-overrides . Tunnel address selection informs the client which address to reserve. For reference 

https://github.com/opnsense/core/blob/7c3f2de92da8c2174578650957c643d6ced386e2/src/etc/inc/plugins.inc.d/openvpn.inc#L983-L993

**Comment by muchachagrande** [comment]
So, to have it clear, from what I see in the code, the IP set in tunnel_network will be pushed to the client in the case that we are not using topology_subnet. Isn't it?

**Comment by AdSchellevis** [comment]
it always pushes the address, how just depends on the type

**Comment by muchachagrande** [comment]
Thank you @AdSchellevis. I've looked for this on the forum and just found unanswered questions.

May be, modifying the help text a bit to include both use cases would help others, being that the label "Tunnel Network" is not that descriptive of the functionality.

I've tested it and work as expected, so I'll close this issue.
Thank you again for your time and help. It was the fastest feature addition ever made. :joy:
